### DISCUSSION TITLE
How to become root on a non-web ddev container

### DISCUSSION THREAD

**Discussion by apotek** [ddev/ddev#6289] [General]
While reading [the docs about how to customize my ddev containers](https://ddev.readthedocs.io/en/stable/users/extend/customizing-images/#adding-extra-debian-packages-with-webimage_extra_packages-and-dbimage_extra_packages), I saw this:

> Debugging the Dockerfile Build¶
> 
> It can be complicated to figure out what’s going on when building a Dockerfile, and even more complicated when you’re seeing it go by as part of ddev start.
> 
>  Use ddev ssh first of all to pioneer the steps you want to take. You can do all the things you need to do there and see if it works.

The web image ships with `sudo` so this is certainly doable there. But what about the db container? 

```
% ddev ssh -s db
me@my-project-db:~$ sudo apt-get update
bash: sudo: command not found
```

Ok. Same thing happens even when accessing the container directly via docker:

```
docker exec -it <CONTAINERID> bash
me@my-project-db:~$ sudo 
bash: sudo: command not found
```

So there doesn't really seem to be any way to experiment with extending or customizing this container. Is the lack of sudo an oversight or a result of default security models on a database container?

**Comment by rfay** [comment]
DDEV's ddev-dbserver does not ship with sudo for security reasons, but you can add it (and it's preconfigured) if you use `dbimage_extra_packages: [sudo]`

You can also get into most containers with `docker exec -it -u root <containername>` sh`, for example `docker exec -it -u root ddev-d10-db sh`

**Reply by apotek** [reply]
Thank you! These are both smart solutions! passing `-u root` to docker exec seems like the better way, since it doesn't modify the image permanently. ++

**Reply by rfay** [reply]
What is it that you need to do in the ddev-dbserver? Not many people fiddle there.
